### PR TITLE
feat: reorg test bridge

### PR DESCRIPTION
**Pull request by arnaubennassar**
- Added halt behaviour on the bridgesync: now it's capable to detect data inconsistencies due to unexpected deposit count on the tree. When this happens the syncer halts. A reorg on the DB could unhalt the syncer
- Other minor fixes related to race condition and resillience
- Changed the config for the sqlite DB connections
- Add reorg test for bridgesync. Have successfully runned the test with 1ms block times and the following config:

```go
func TestBridgeEventE2E(t *testing.T) {
	const (
		totalBridges          = 1000
		totalReorgs           = 700
		maxReorgDepth         = 5
		reorgEveryXIterations = 7 // every X blocks go back [1,maxReorgDepth] blocks
	)
```

test took 235.249s, ~3k blocks were synced, the bridge sync only perceived 96/700 reorgs due to the speed in which blocks are created and reorged with this config. In the final config, the test made the syncers process 10 reorgs in ~13s

### CODE REVIEW

File: test/helpers/reorg.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+func Reorg(t *testing.T, client *simulated.Backend, reorgSizeInBlocks uint64) {
+	t.Helper()
+	ctx := context.Background()
+	currentBlockNum, err := client.Client().BlockNumber(ctx)
+	require.NoError(t, err)
+
+	block, err := client.Client().BlockByNumber(ctx, big.NewInt(int64(currentBlockNum-reorgSizeInBlocks)))
+	log.Debugf("reorging until block %d. Current block %d (before reorg)", block.NumberU64(), currentBlockNum)
```
Comment (goran-ethernal) [inline test/helpers/reorg.go:31]: Move this after the require no error.

**Review by goran-ethernal — APPROVED** [review]

**Review by vcastellm — APPROVED** [review]
LGTM

**Review by joanestebanr — CHANGES_REQUESTED** [review]
why do you need a real file on unittest?

File: aggsender/db/aggsender_db_storage_test.go [hunk 2]
```diff
@@ -20,7 +20,7 @@ import (
 func Test_Storage(t *testing.T) {
 	ctx := context.Background()
 
-	path := path.Join(t.TempDir(), "file::memory:?cache=shared")
+	path := path.Join(t.TempDir(), "aggsenderTest_Storage.sqlite")
```
[the remark this answers is not in the document]
Comment (goran-ethernal) [reply aggsender/db/aggsender_db_storage_test.go:23]: I've seen this in the aggsender tests, that is why I've added that clean function.

[the remark this answers is not in the document]
Comment (arnaubennassar) [reply aggsender/db/aggsender_db_storage_test.go:23]: yes, I faced problems making things work in memory in the past + it was already using file!

Other than that, let's try to have as realistic tests as possible. It's ok to mock when it facilitates testing, but in this case, having a more realistic setup is super easy and doesn't add friction. The way sqlite is configured has a huge impact on how it works: some tests broke when I changed the config, so I believe we should aim to have the same setup on tests vs on prod

**Review by joanestebanr — APPROVED** [review]

**Pull request merged**